src/Orchestration/Adapter/DockerAPI.php

```diff
@@ -0,0 +1,380 @@
+            'Content-Type: application/json',
+            'Content-Length: ' . \strlen(\json_encode($body))
+        ]);
+
+        if ($result['code'] !== 201) {
+            throw new DockerAPIException('Failed to create function environment: '. $result['response']. ' Response Code: '. $result['code']);
+        }
+
+        $parsedResponse = json_decode($result['response'], true);
+
+        // Run Created Container
+        $result = $this->call('http://localhost/containers/'.$parsedResponse['Id'].'/start', 'POST', '{}');
+        
+        if ($result['code'] !== 204) {
+            throw new DockerAPIException('Failed to create function environment: '.$result['response'].' Response Code: '.$result['code']);
+        } else {
+            return $parsedResponse['Id'];
+        }
+    }
+
+    /**
+     * Execute Container
+     *
+     * @param string $name
+     * @param string[] $command
+     * @param string &$stdout
+     * @param string &$stderr
+     * @param array<string, string> $vars
+     * @param int $timeout
+     * @return bool
+     */
+    public function execute(string $name, array $command, string &$stdout, string &$stderr, array $vars = [], int $timeout = -1): bool
+    {
+        foreach ($vars as $key => &$value) {
+            $key = $this->filterEnvKey($key);
+            $value = $key.'='.$value;
+        }
+
+        $body = [
+            'Env' => \array_values($vars),
+            'Cmd' => $command,
+            'AttachStdout' => true,
+            'AttachStderr' => true
+        ];
+
+        $result = $this->call('http://localhost/containers/'.$name.'/exec', 'POST', json_encode($body), [
+            'Content-Type: application/json',
+            'Content-Length: ' . \strlen(\json_encode($body)),
+        ], $timeout);
+
+        if ($result['code'] !== 201) {
+            throw new DockerAPIException('Failed to create execute command: '.$result['response'].' Response Code: '.$result['code']);
+        }
+
+        $parsedResponse = json_decode($result['response'], true);
+
+        $result = $this->streamCall('http://localhost/exec/'.$parsedResponse['Id'].'/start', $timeout);
+
+        $stdout = $result['stdout'];
+        $stderr = $result['stderr'];
+
+        if ($result['code'] !== 200) {
+            throw new DockerAPIException('Failed to create execute command: '.$result['response'].' Response Code: '. $result['code']);
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Remove Container
+     *
+     * @param string $name
+     * @param bool $force
+     * @return bool
+     */
+    public function remove(string $name, bool $force = false): bool
+    {
+        $result = $this->call('http://localhost/containers/'.$name.($force ? '?force=true': ''), 'DELETE');
+
+        if ($result['code'] !== 204) {
+            throw new DockerAPIException('Failed to remove container: '.$result['response'].' Response Code: '.$result['code']);
```

Comment:
Not sure if throwing an exception is needed, since the method will return false on failure.

Reply:
Probably better to rely on the exception so we can pass the error data, and report it to the end-developer.